grpcserver.go

```diff
@@ -3,22 +3,29 @@ package main
 import (
 	"context"
 	"errors"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
 	"github.com/gcash/bchd/bchrpc"
+	"github.com/gorilla/mux"
+	grpc_prometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
 	"github.com/improbable-eng/grpc-web/go/grpcweb"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
 	"google.golang.org/grpc/metadata"
 	"google.golang.org/grpc/peer"
-	"net"
-	"net/http"
-	"strings"
 )
 
 // AuthenticationTokenKey is the key used in the context to authenticate clients.
 // If this is set to anything other than "" in the config, then the server expects
 // the client to set a key value in the context metadata to 'AuthenticationToken: cfg.AuthToken'
 const AuthenticationTokenKey = "AuthenticationToken"
 
+var prometheusEnabled = false
```

Comment:
the logging is nice. But this line is ugly. Variables shouldn't be set globally in a file. Instead bound to a struct (such as `rpcCfg *bchrpc.GrpcServerConfig`)

Reply:
So your original PR broke all RPC calls if prometheus wasn't specified, plus those interceptors don't have access to the config. If you have a better pattern to check if prometheus is enabled, I am happy to change it.

Reply:
Doing a cast from `interface{}`

Reply:
So you prefer doing a type check on every grpc request vs a package level var that just tracks if prometheus has been enabled? I agree that globals aren't great, but doing type checks on every grpc request seems expensive...

Reply:
Hmm agreed - didn't benchmark and generally type assertions are fast in Go (unlike type switch). But still even without benchmark we can agree your solution is faster and less code. We won't start a 2nd gRPC server ever where the global gets in the way I guess